frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -372,57 +416,87 @@ pub mod pallet {
         /// This must be called by the developer who registered the contract.
         ///
         /// Warning: After this action contract can not be assigned again.
-        #[pallet::weight(T::WeightInfo::unregister(T::MaxNumberOfStakersPerContract::get()))]
+        #[pallet::weight(T::WeightInfo::unregister())]
         pub fn unregister(
             origin: OriginFor<T>,
             contract_id: T::SmartContract,
         ) -> DispatchResultWithPostInfo {
-            ensure!(!Self::pallet_disabled(), Error::<T>::Disabled);
-            let developer = ensure_signed(origin)?;
-
-            let registered_contract =
-                RegisteredDevelopers::<T>::get(&developer).ok_or(Error::<T>::NotOwnedContract)?;
+            Self::ensure_pallet_enabled()?;
```

Comment:
I'm not sure that root calls should have this guard

Reply:
Btw, perhaps looking at the test utility method for `withdraw_from_unregistered` would help. It think it shows how behavior is different.